### ISSUE TITLE
Compact database commands create unwanted subdirectories

### ISSUE THREAD

**Issue by hvredevoort** [CPChain/chain#97]
**Where does the bug occur?**
[e.g. the chain, the wallet, the explorer]
chain

**What is your wallet address and your balance?**
If it is too private to answer, you can skip it. 
irrelevant

**Describe the bug**
When the following command is given in this or other variations:
./cpchain chain delete dpor- --datadir ~/.cpchain
./cpchain chain delete dpor- --datadir ~/cpchain
./cpchain chain delete dpor- --datadir /home/username/cpchain
./cpchain chain delete dpor- --datadir /home/username/cpchain/datadir
./cpchain chain delete dpor- --datadir /home/username/cpchain/datadir/chaindata

it creates several different directories like cpchain, keystore, chaindata below the ./datadir directory or other directories depending on what parameter is used.
If the default is used with ~/.cpchain it also creates these directories below that directory.
If you then issue du -h --total it will have no permission to read those directories.

**To Reproduce**
Steps to reproduce the behavior, and the commands you use.
Try several parameters for --datadir
./cpchain chain delete dpor- --datadir ~/.cpchain
./cpchain chain delete dpor- --datadir ~/cpchain
./cpchain chain delete dpor- --datadir /home/username/cpchain
./cpchain chain delete dpor- --datadir /home/username/cpchain/datadir
./cpchain chain delete dpor- --datadir /home/username/cpchain/datadir/chaindata

**Expected behavior**
I don't expect directories to be created, this is a bug.

**Screenshots**

** Configuration (please complete the following information):**
 - OS: Ubuntu 18.10
 - cpchain version 0.5.1

**Additional context**
If your bug can only be reproduced occasionally, please emphasize it.

**Comment by hvredevoort** [comment]
Please test this thoroughly as it can really mess up your RNode.
Also provide an example of the correct --datadir parameter if the cpchain program is in ~/cpchain and datadir is below ~/cpchain

**Comment by alex031029** [comment]
Your report confused me a lot. I called the guy that developed the compact feature, and we reviewed this issue, and currently have no idea why it was that messy and why you had no permission of these subdirectories. 

We will try to reproduce your issue, and figure it out.

**Comment by alex031029** [comment]
Ah, I got it. The reason why your subdirectories are messy is that the `--datadir` flag is referring to whole cpchain directory which includes `keystore` and `cpchain` etc. Here I attached the directory structure of my computer.

![image](https://user-images.githubusercontent.com/2452221/69840183-90fabd00-1295-11ea-8b7e-8dd218caed1a.png)


If `--datadir` is not referring the whole cpchain directory (as in my case, the /home/shi/.cpchain), the compact command will create subdirectories like `cpchain` and `keystore` under it.
